test/f90_correct/src/io28.f90

```diff
@@ -0,0 +1,38 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! test for I/O take derived types
+
+program test
+  type my_type
+     integer :: i
+     integer :: j
+     integer :: k
+  end type my_type
+  character(80) :: str1, str2
+  type(my_type), dimension(3) :: t
```

Comment:
Why do you need three elements in this array? Wouldn't just two suffice to show the problem?

Reply:
Flang would call incorrectly `func()` tree time and write the value of `t(1)%i`, `t(2)%j` and `t(3)%k` to `str2`, so I have declared three elements in this array. But it need only one element if the issue is fixed.

Reply:
> Why do you need three elements in this array? Wouldn't just two suffice to show the problem?

Flang would call incorrectly func() tree time and write the value of t(1)%i, t(2)%j and t(3)%k to str2, so I have declared three elements in this array. But it need only one element if the issue is fixed.

Reply:
OK thanks.

Reply:
Can you check the simplified test case as the following?
```
program test
  type my_type
     integer :: i
     integer :: j
  end type my_type
  character(80) :: str1(2), str2(2)
  type(my_type) :: t(2)

  t(1) = my_type(11, 12)
  t(2) = my_type(21, 22)

  write(str1(1), *), t(1)
  write(str1(2), *), t(2)
  write(str2(1), *), t(func())
  write(str2(2), *), t(func())
  if (any(str1 .ne. str2)) STOP 1
  write(*, *) 'PASS'

contains
  integer function func()
    integer, save :: i = 1
    func = i
    i = i + 1
  end function
end
```

Reply:
OK. Thank you.